sqlmesh/core/context_diff.py

```diff
@@ -125,8 +127,11 @@ def create(
             modified = modified_info.get(name)
             existing = stored.get(snapshot.snapshot_id)
 
-            if existing:
-                # Keep the original model instance to preserve the query cache.
+            if modified and snapshot.node_type != modified.node_type:
```

Comment:
Shouldn't we remove it from the `modified` mapping then?

Reply:
modified isn't used again, but I can remove it to prevent bugs in future modifications.